src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerHelper.cs

```diff
@@ -0,0 +1,113 @@
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Helper class for OData Serialization
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1053:StaticHolderTypesShouldNotHaveConstructors")]
+    internal class ODataSerializerHelper
+    {
+        /// <summary>
+        /// Appends instance annotations to an ODataResource.
+        /// </summary>
+        /// <param name="resource">The ODataResource being annotated</param>
+        /// <param name="resourceContext">The context for the resource instance to be annotated.</param>
+        /// <param name="value">The annotations to write.</param>
+        /// <param name="serializerProvider">The SerializerProvider to use to write annotations</param>
+        internal static void AppendInstanceAnnotations(ODataResourceBase resource, ResourceContext resourceContext, object value, ODataSerializerProvider serializerProvider)
+        {
+            IODataInstanceAnnotationContainer instanceAnnotationContainer = value as IODataInstanceAnnotationContainer;
+
+            if (instanceAnnotationContainer == null)
+            {
+                return;
+            }
+
+            IDictionary<string, object> clrAnnotations = instanceAnnotationContainer.GetResourceAnnotations();
+
+            if (clrAnnotations != null)
+            {
+                foreach (KeyValuePair<string, object> annotation in clrAnnotations)
+                {
+                    AddODataAnnotations(resource.InstanceAnnotations, resourceContext, annotation, serializerProvider);
+                }
+            }
+
+            if (resource.Properties != null)
+            {
+                foreach (ODataProperty property in resource.Properties)
+                {
+                    string propertyName = property.Name;
+
+                    IDictionary<string, object> propertyAnnotations = instanceAnnotationContainer.GetPropertyAnnotations(propertyName);
+
+                    if (propertyAnnotations != null)
+                    {
+                        foreach (KeyValuePair<string, object> annotation in propertyAnnotations)
+                        {
+                            AddODataAnnotations(property.InstanceAnnotations, resourceContext, annotation, serializerProvider);
+                        }
+                    }
+                }
+            }
+        }
+
+        internal static void AddODataAnnotations(ICollection<ODataInstanceAnnotation> InstanceAnnotations, ResourceContext resourceContext, KeyValuePair<string, object> annotation, ODataSerializerProvider serializerProvider)
```

Comment:
Should this method be `internal` or `private`? If `internal` please document the method and the parameters

Reply:
its private